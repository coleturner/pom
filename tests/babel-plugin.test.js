import pluginTester from 'babel-plugin-tester';
import plugin from '../src/babel/plugin-7';

pluginTester({
  plugin,
  snapshot: true,
  tests: [
    // Standard, full dead code elimination
    `
        import { createSelector } from 'pom';
        const selector = createSelector({ prefix: 'cool '});
      `,
    `
        import POM from 'pom';
        const selector = POM.createSelector({ prefix: 'cool '});
    `,
    `
        const { createSelector } = require('pom');
        const selector = createSelector({ prefix: 'cool '});
    `,
    `
        const POM = require('pom');
        const selector = POM.createSelector({ prefix: 'cool '});
    `,

    // Leftover imports
    `
        import POM, { createSelector, somethingElse } from 'pom';
        const selector = createSelector({ prefix: 'cool '});
        POM.keepsThis();
        POM.keepsThat();
        somethingElse('stays too');
    `,
    `
        const { createSelector, somethingElse } = require('pom');
        const selector = createSelector({ prefix: 'cool '});
        somethingElse('stays put');
    `,
    `
        const { createSelector, somethingElse } = require('pom');
        const selector = createSelector({ prefix: 'cool '});
        somethingElse('stays put');
        somethingElse('stays put');
    `,
  ],
});
