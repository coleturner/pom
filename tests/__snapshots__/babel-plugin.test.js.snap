// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`solidselectors 1. solidselectors: 1. solidselectors 1`] = `
"
import { createSelector } from 'solidselectors';
const selector = createSelector({ prefix: 'cool '});

      ↓ ↓ ↓ ↓ ↓ ↓

const selector = \\"cool :c0642c2f-5c84-4f61-b338-2b813887d53f\\";
"
`;

exports[`solidselectors 2. solidselectors: 2. solidselectors 1`] = `
"
import POM from 'solidselectors';
const selector = POM.createSelector({ prefix: 'cool '});

      ↓ ↓ ↓ ↓ ↓ ↓

const selector = \\"cool :43d2f1c8-e680-434c-bc1e-ff083ba4e5de\\";
"
`;

exports[`solidselectors 3. solidselectors: 3. solidselectors 1`] = `
"
const { createSelector } = require('solidselectors');
const selector = createSelector({ prefix: 'cool '});

      ↓ ↓ ↓ ↓ ↓ ↓

const selector = \\"cool :534b866f-1ece-4cce-8e4c-de9c2dc660c0\\";
"
`;

exports[`solidselectors 4. solidselectors: 4. solidselectors 1`] = `
"
const POM = require('solidselectors');
const selector = POM.createSelector({ prefix: 'cool '});

      ↓ ↓ ↓ ↓ ↓ ↓

const selector = \\"cool :d11b1661-83b6-4aa7-99b0-6cf921471834\\";
"
`;

exports[`solidselectors 5. solidselectors: 5. solidselectors 1`] = `
"
import POM, { createSelector, somethingElse } from 'solidselectors';
const selector = createSelector({ prefix: 'cool '});
POM.keepsThis();
POM.keepsThat();
somethingElse('stays too');

      ↓ ↓ ↓ ↓ ↓ ↓

import POM, { somethingElse } from 'solidselectors';
const selector = \\"cool :932cc4f4-1dc1-4a95-b874-7d2347cfda0d\\";
POM.keepsThis();
POM.keepsThat();
somethingElse('stays too');
"
`;

exports[`solidselectors 6. solidselectors: 6. solidselectors 1`] = `
"
const { createSelector, somethingElse } = require('solidselectors');
const selector = createSelector({ prefix: 'cool '});
somethingElse('stays put');

      ↓ ↓ ↓ ↓ ↓ ↓

const {
  somethingElse
} = require('solidselectors');

const selector = \\"cool :b649f30f-0306-4296-974b-fffd75dbf9d8\\";
somethingElse('stays put');
"
`;

exports[`solidselectors 7. solidselectors: 7. solidselectors 1`] = `
"
const { createSelector, somethingElse } = require('solidselectors');
const selector = createSelector({ prefix: 'cool '});
somethingElse('stays put');
somethingElse('stays put');

      ↓ ↓ ↓ ↓ ↓ ↓

const {
  somethingElse
} = require('solidselectors');

const selector = \\"cool :f1b9cb6e-fed6-4ca6-b8b6-d761bbc87b63\\";
somethingElse('stays put');
somethingElse('stays put');
"
`;
